src/Step/FilterStep.php

```diff
@@ -39,7 +40,7 @@ public function process($item, callable $next)
     {
         foreach (clone $this->filters as $filter) {
             if (false === call_user_func($filter, $item)) {
-                return false;
+                throw new FilterException();
```

Comment:
I am not sure, I can't think of any clean solution, only ugly ones (with variables passed by reference at some point)